code/web/src/modules/user/UserInfo.js

```diff
@@ -1,249 +1,304 @@
-                    <H5>Shipping Address</H5>
-                    <Input
-                        data-testid='shipping-input'
-                        type='text'
-                        name='shippingAddressInput'
-                        value={this.state.shippingAddressInput}
-                        style={shippingInput}
-                        onChange={(e) => this.handleChange(e)}
-                    />
-                </section>
-            </section>
-        );
-    };
-    renderInfo = () => {
-        return (
-            <section style={infoContainer}>
-                <section style={imageEmailNameWrapper}>
-                    <div style={imageWrapper}>
-                        <img
-                            src={this.state.imageUrl}
-                            style={imageStyle}
-                            data-testid='user-img'
-                            alt='user image'
-                        />
-                    </div>
-                    <section style={emailNameWrapper}>
-                        <H3>{this.props.user.details.name}</H3>
-                        <p style={emailInput}>{this.state.emailInput}</p>
-                    </section>
-                </section>
-                <section style={descriptionWrapper}>
-                    <H3>About me</H3>
-                    <p style={descriptionInput}>{this.state.userDescription}</p>
-                </section>
-                <section style={shippingAddress}>
-                    <H5>Shipping Address</H5>
-                    <p style={shippingInput}>
-                        {this.state.shippingAddressInput}
-                    </p>
-                </section>
-            </section>
-        );
-    };
-    render() {
-        return (
-            <section style={style}>
-                {this.state.isEditing ? (
-                    <button
-                        onClick={this.toggleEdit}
-                        style={editSaveButton}
-                        data-testid='editUserBtn'
-                    >
-                        Save Changes
-                    </button>
-                ) : (
-                    <button
-                        onClick={this.toggleEdit}
-                        style={editSaveButton}
-                        data-testid='editUserBtn'
-                    >
-                        Edit
-                    </button>
-                )}
-                {this.state.isEditing ? this.renderInputs() : this.renderInfo()}
-            </section>
-        );
-    }
+          let user = this.state
+          user.imageUrl = `/images/uploads/${response.data.file}`
+
+          this.setState({
+            ...user,
+          })
+        } else {
+          this.props.messageShow('Please try again.')
+        }
+      })
+      .catch((error) => {
+        this.props.messageShow('There was some error. Please try again.')
+      })
+      .then(() => {
+        window.setTimeout(() => {
```

Comment:
Usually dislike setTimeout but this is a great implementation!

Reply:
What does this do?